### PR TITLE
[MRG] Rework dependency specification

### PR DESCRIPTION
**Pull request by hoechenberger**
Dependencies should be specified in `setup.py`, otherwise `pip` won't know about them.

I've added the basic requirements – numpy, scipy, mne – as `install_requires`. Additional deps – nibabel, pybv, matplotlib –
are an `extra` now and can be installed via

```
pip install mne-bids[full]
```
The `requirements.txt` file was only used for testing, so I renamed it to `test_requirements.txt`

I've also simplified the installation instructions.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: doc/install.rst [hunk 1]
```diff
@@ -28,45 +28,36 @@ You may choose to install ``mne-bids``
...
 
-If you want to use the latest development version of ``mne-bids``, use the
-following command:
+MNE-BIDS works best with the latest stable release of MNE-Python. To ensure
+MNE-Python is up-to-date, run:
 
 .. code-block:: bash
 
-   $ pip install https://api.github.com/repos/mne-tools/mne-bids/zipball/master
+   pip install -U mne
```
Comment (agramfort) [inline doc/install.rst:60]: why did you remove this?

Comment (hoechenberger) [reply doc/install.rst:60]: I think if anything, it should go to the developer installation instructions, no? Normal users shouldn't have to run `master` – if they do, we should probably create a new release :)

**Comment by agramfort** [comment]
well we cannot release every other day and it helps regularly users. I
would keep it

>

**Comment by hoechenberger** [comment]
> well we cannot release every other day and it helps regularly users. I would keep it

Ok I've added it back again!

**Comment by hoechenberger** [comment]
Updated install instructions:
https://3659-89170358-gh.circle-artifacts.com/0/dev/install.html

**Review by agramfort — APPROVED** [review]
thx @hoechenberger

File: doc/install.rst [hunk 2]
```diff
@@ -28,45 +28,42 @@ You may choose to install ``mne-bids``
...
+This ``pip`` command will also work if you want to upgrade if a newer version
+of ``mne-bids`` is available.
 
-Then install ``mne-bids``\ :
+If you don't require advanced features like interactive visual data inspection,
+you may also install a basic version of MNE-BIDS via
 
 .. code-block:: bash
 
-   $ pip install -U mne-bids
```
Comment (sappelhoff) [inline doc/install.rst:44]: does the removal of the `$` break the recently added copybutton functionality? see line 45 in conf.py

Comment (sappelhoff) [reply doc/install.rst:44]: nope, just checked.

**Review by sappelhoff — APPROVED** [review]

File: setup.py [hunk 3]
```diff
@@ -38,6 +38,18 @@
...
+          install_requires=[
+              'mne >=0.21',
+              'numpy >=1.14',
+              'scipy >=0.18.1',
+          ],
+          extras_require={
+              'full': [
+                  'nibabel >=2.2',
+                  'pybv >=0.4',
+                  'matplotlib'
```
Comment (sappelhoff) [inline setup.py:50]: why is pandas not listed here, but it's listed as an optional dependency in the installation instructions?

Comment (hoechenberger) [reply setup.py:50]: Good question! I will add it.

**Comment by hoechenberger** [comment]
Added pandas.

@sappelhoff Feel free to merge once CI goes green.

**Comment by hoechenberger** [comment]
seems I broke something =)

File: setup.py [hunk 4]
```diff
@@ -38,6 +38,19 @@
           long_description=open('README.md').read(),
           long_description_content_type='text/markdown',
           python_requires='~=3.6',
+          install_requires=[
+              'mne >=0.21',
+              'numpy >=1.14',
+              'scipy >=0.18.1',
+          ],
```
Comment (jasmainak) [inline setup.py:45]: Is this also going to install these or upgrade? I personally don't like it when packages do this. It messes with the environment they have.

Comment (hoechenberger) [reply setup.py:45]: If these requirements are not met, they will be installed.

If you don't like this, simply run `pip` with the `--no-deps` switch.

**Comment by hoechenberger** [comment]
@sappelhoff We're green! :)

**Comment by sappelhoff** [comment]
Thanks @hoechenberger
